test/e2e/overrides_test.go

```diff
@@ -140,6 +143,168 @@ var _ = Describe("Overrides test", Ordered, func() {
+			err = verifyOperatorStatusCondition(certmanageroperatorclient, []string{certManagerCAInjectorDeploymentControllerName}, validOperatorStatusConditions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the resources to be added to the cert-manager cainjector deployment")
+			err = verifyDeploymentResources(k8sClientSet, certmanagerCAinjectorDeployment, res, true)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("When adding invalid cert-manager controller override resources", func() {
+
+		It("should not add the resources to the cert-manager controller deployment", func() {
+
+			By("Adding cert-manager controller override resources to the certmanagaer.operator object")
+			res := v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("2Gi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("1Gi"),
+				},
+			}
+			err := addOverrideResources(certmanageroperatorclient, certmanagerControllerDeployment, res)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager controller status to become degraded")
+			err = verifyOperatorStatusCondition(certmanageroperatorclient, []string{certManagerControllerDeploymentControllerName}, invalidOperatorStatusConditions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking if the resources are not added to the cert-manager controller deployment")
+			err = verifyDeploymentResources(k8sClientSet, certmanagerControllerDeployment, res, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("When adding invalid cert-manager webhook override resources", func() {
+
+		It("should not add the resources to the cert-manager webhook deployment", func() {
+
+			By("Adding cert-manager webhook override resources to the certmanagaer.operator object")
+			res := v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("2Gi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("1Gi"),
+				},
+			}
+			err := addOverrideResources(certmanageroperatorclient, certmanagerWebhookDeployment, res)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager webhook controller status to become degraded")
+			err = verifyOperatorStatusCondition(certmanageroperatorclient, []string{certManagerWebhookDeploymentControllerName}, invalidOperatorStatusConditions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking if the resources are not added to the cert-manager webhook deployment")
+			err = verifyDeploymentResources(k8sClientSet, certmanagerWebhookDeployment, res, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("When adding invalid cert-manager cainjector override resources", func() {
+
+		It("should not add the resources to the cert-manager cainjector deployment", func() {
```

Comment:
Just a comment placeholder as tracker: as we talked offline, let's combine multiple `It`s into one single `It`, corresponding to the single test case describe in CM-230.